ofrak_patch_maker/ofrak_patch_maker/patch_maker.py

```diff
@@ -229,43 +237,81 @@ def make_bom(
-            assembled_object_path = self._toolchain.compile(c_file, header_dirs, out_dir=out_dir)
-            assembled_object = self.prepare_object(assembled_object_path)
-            object_map.update({c_file: assembled_object})
-
-        for asm_file in list(filter(lambda x: x.endswith(".as") or x.endswith(".S"), source_list)):
-            original_asm_file = asm_file
-            if asm_file.endswith(".S"):
-                asm_file = self._toolchain.preprocess(asm_file, header_dirs, out_dir=out_dir)
-            assembled_object_path = self._toolchain.assemble(asm_file, header_dirs, out_dir=out_dir)
-            assembled_object = self.prepare_object(assembled_object_path)
-            object_map.update({original_asm_file: assembled_object})
+        # Compile .c files in parallel
+        c_files = list(filter(lambda x: x.endswith(".c"), source_list))
+        c_args = zip(
+            c_files,
+            itertools.repeat(header_dirs),
+            itertools.repeat(out_dir),
+            itertools.repeat(self._toolchain),
+            itertools.repeat(self.prepare_object),
+            itertools.repeat(self),
+            itertools.repeat(SourceFileType.C),
+        )
+        workers = math.ceil(cpu_count())
+        with get_context("spawn").Pool(processes=workers) as pool:
+            result = pool.starmap(
+                _create_object_file, c_args, chunksize=math.ceil(len(c_files) / workers)
+            )
+        for r in result:
+            object_map.update(r)
+
+        # Assemble .asm files in parallel
+        asm_files = list(filter(lambda x: x.endswith(".as") or x.endswith(".S"), source_list))
+        asm_args = zip(
+            asm_files,
+            itertools.repeat(header_dirs),
+            itertools.repeat(out_dir),
+            itertools.repeat(self._toolchain),
+            itertools.repeat(self.prepare_object),
+            itertools.repeat(self),
+            itertools.repeat(SourceFileType.ASM),
+        )
+        workers = math.ceil(cpu_count())
+        with get_context("spawn").Pool(processes=workers) as pool:
+            result = pool.starmap(
+                _create_object_file, asm_args, chunksize=math.ceil(len(asm_files) / workers)
+            )
+        for r in result:
+            object_map.update(r)
 
         # Compute the required size for the .bss segment
         bss_size_required = 0
         symbols: Dict[str, int] = {}
+        unresolved_symbols: Set[str] = set()
         for o in object_map.values():
             bss_size_required += o.bss_size_required
             symbols.update(o.symbols)
+            # Resolve symbols defined within different patch files within the same patch BOM
+            for sym in o.rel_symbols.keys():
+                # Have already seen this symbol in a previous patch object
+                if sym in symbols.keys():
+                    continue
+                elif sym in unresolved_symbols:
```

Comment:
Why is this case here? Isn't `sym` an unresolved symbol at this point? 

If I understand correctly, if it's in `unresolved_symbols`, that means it was also an unresolved symbol used in another object file. Is that correct? If so, then removing it from `unresolved_symbols` is not correct behavior.